test/syspurpose/base.py

```diff
@@ -11,145 +11,9 @@
-            self.fail(err_msg)
-
-    # Taken from subman test Fixtures
-    def assert_equal_dict(self, expected_dict, actual_dict):
-        mismatches = []
-        missing_keys = []
-        extra = []
-
-        for key in expected_dict:
-            if key not in actual_dict:
-                missing_keys.append(key)
-                continue
-            if expected_dict[key] != actual_dict[key]:
-                mismatches.append((key, expected_dict[key], actual_dict[key]))
-
-        for key in actual_dict:
-            if key not in expected_dict:
-                extra.append(key)
-
-        message = ""
-        if missing_keys or extra:
-            message += "Keys in only one dict: \n"
-            if missing_keys:
-                for key in missing_keys:
-                    message += "actual_dict:  %s\n" % key
-            if extra:
-                for key in extra:
-                    message += "expected_dict: %s\n" % key
-        if mismatches:
-            message += "Unequal values: \n"
-            for info in mismatches:
-                message += "%s: %s != %s\n" % info
-
-        # pprint the dicts
-        message += "\n"
-        message += "expected_dict:\n"
-        message += pprint.pformat(expected_dict)
-        message += "\n"
-        message += "actual_dict:\n"
-        message += pprint.pformat(actual_dict)
-
-        if mismatches or missing_keys or extra:
-            self.fail(message)
-
-
-class Capture(object):
-    class Tee(object):
-        def __init__(self, stream, silent):
-            self.buf = io.StringIO()
-            self.stream = stream
-            self.silent = silent
-
-        def write(self, data):
-            self.buf.write(data)
-            if not self.silent:
-                self.stream.write(data)
-
-        def flush(self):
-            pass
-
-        def getvalue(self):
-            return self.buf.getvalue()
-
-        def isatty(self):
-            return False
-
-    def __init__(self, silent=False):
-        self.silent = silent
-
-    def __enter__(self):
-        self.buffs = (self.Tee(sys.stdout, self.silent), self.Tee(sys.stderr, self.silent))
-        self.stdout = sys.stdout
-        self.stderr = sys.stderr
-        sys.stdout, sys.stderr = self.buffs
-        return self
-
-    @property
-    def out(self):
-        return self.buffs[0].getvalue()
-
-    @property
-    def err(self):
-        return self.buffs[1].getvalue()
-
-    def __exit__(self, exc_type, exc_value, traceback):
-        sys.stdout = self.stdout
-        sys.stderr = self.stderr
-
-
-def write_to_file_utf8(file, data):
-    """
-    Writes out the provided data to the specified file, with user-friendly indentation,
-    and in utf-8 encoding.
-    :param file: The file to write to
-    :param data: The data to be written
-    :return:
-    """
-    file.write(json.dumps(data, indent=2, ensure_ascii=False))
+    pass
```

Comment:
> syspurpose tests can still get common functionalities *in case they are needed*.

How likely is the scenario that we'll need this class? Can't we re-add the class if that happens?

Reply:
Hm you are right, it can be added again (or done differently) if needed. Let's drop this then.